boot-speed/testflinger/measure-device.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+
+set -eux
+set -o pipefail
+shopt -s nullglob
+
+if [ $# -ne 2 ]; then
+    echo "Usage: $0 <device> <distro>"
+    exit 1
+fi
+
+# Are the required command available? Fail early if they are not.
+# We rely on the errexit (set -e) option here.
+command -v testflinger-cli
+command -v timeout
+command -v jq
+
+device=$1
+distro=$2
+
+yaml_head="$device-$distro.provision.yaml"
+yaml_tail="test_data.yaml"
+yaml_full="$device-$distro.full.yaml"
+
+yyyymmdd=$(date --utc '+%Y%m%d')
+date_rfc3339=$(date --utc --rfc-3339=ns)
+
+echo "Target device: $device"
+echo "Target distribution: $distro"
+
+datadir="$device-${distro}_$yyyymmdd"
+mkdir -v $datadir
+
+image_url=$(grep "url:" "$yaml_head" | awk '{ print $2 }')
+image_dirname=$(echo $image_url | sed 's|\(.*/\)\(.*\)|\1|')
+image_basename=$(echo $image_url | sed 's|\(.*/\)\(.*\)|\2|')
+image_serial=$(curl -s --noproxy ubuntu.com $image_dirname/.publish_info | grep $image_basename | awk '{ print $2 }') || true
+
+regexp="^[0-9]{8}(\.[0-9]{1,2})?$"
+if [[ ! "$image_serial" =~ $regexp ]]; then
+    echo "WARNING: Image serial not found! Falling back using today's date."
```

Comment:
Workaround to be removed after the .publish_info files will be in place for Core 16 and Core 18.